lmod_manager/lmod_manager.py

```diff
@@ -20,8 +21,10 @@
 from subprocess import call
 from typing import Optional, Union
 
-DEFAULT_LMOD_MODULES_DIR = "/etc/lmod/modules"
-DEFAULT_INSTALLATION_DIR = "/opt"
+RUN_AS_ROOT = os.geteuid() == 0
+HOME = os.getenv("HOME")
+DEFAULT_LMOD_MODULES_DIR = "/etc/lmod/modules" if RUN_AS_ROOT else f"{HOME}/.config/lmod/modulefiles"
+DEFAULT_INSTALLATION_DIR = "/opt" if RUN_AS_ROOT else f"{HOME}/.opt"
```

Comment:
I think it would be a bit nicer to use `pathlib.Path` instead of `os`:

```suggestion
RUN_AS_ROOT = os.geteuid() == 0
HOME = Path.home()
DEFAULT_LMOD_MODULES_DIR = Path("/etc/lmod/modules") if RUN_AS_ROOT else HOME / ".config/lmod/modulefiles"
DEFAULT_INSTALLATION_DIR = Path("/opt") if RUN_AS_ROOT else HOME / ".opt"
```